CodeEdit/Features/Settings/Views/MonospacedFontPicker.swift

```diff
@@ -29,29 +29,41 @@ struct MonospacedFontPicker: View {
         UserDefaults.standard.set(recentFonts, forKey: "recentFonts")
     }
 
-    func getFonts() {
-        DispatchQueue.global(qos: .userInitiated).async {
-            let availableFontFamilies = NSFontManager.shared.availableFontFamilies
+    nonisolated func getFonts() async {
```

Comment:
You're on the right track, but adding `nonisolated` and `aysnc` to this function doesn't make it do it's work on a different thread. `nonisolated` also doesn't do anything to this function because it's not declared on an actor and isn't being called in an isolated context. Instead this should be done in a task group like below. I just used the changes you made here and refactored it into an async context.

```swift
    func getFonts() async {
        await withTaskGroup(of: Void.self) { group in
            group.addTask {
                let monospacedFontFamilyNames = getMonospacedFamilyNames()
                await MainActor.run {
                    self.monospacedFontFamilyNames = monospacedFontFamilyNames
                }
            }

            group.addTask {
                let otherFontFamilyNames = getOtherFontFamilyNames()
                await MainActor.run {
                    self.otherFontFamilyNames = otherFontFamilyNames
                }
            }
        }
    }

    private func getMonospacedFamilyNames() -> [String] {
        let availableFontFamilies = NSFontManager.shared.availableFontFamilies

        return availableFontFamilies.filter { fontName in
            // exclude the font if it is in recentFonts to prevent ForEach conflict
            if recentFonts.contains(fontName) {
                return false
            }

            // exclude default font
            if fontName == "SF Mono" {
                return false
            }

            // include the font which is fixedPitch
            // include the font which numberOfGlyphs is greater than 26
            if let font = NSFont(name: fontName, size: 14) {
                return font.isFixedPitch && font.numberOfGlyphs > 26
            } else {
                return false
            }
        }
    }

    private func getOtherFontFamilyNames() -> [String] {
        let availableFontFamilies = NSFontManager.shared.availableFontFamilies

        return availableFontFamilies.filter { fontName in
            // exclude the font if it is in recentFonts to prevent ForEach conflict
            if recentFonts.contains(fontName) {
                return false
            }

            // include the font which is NOT fixedPitch
            // include the font which numberOfGlyphs is greater than 26
            if let font = NSFont(name: fontName, size: 14) {
                return !font.isFixedPitch && font.numberOfGlyphs > 26
            } else {
                return false
            }
        }
    }
```